### PR TITLE
NetCore: Refactor config/setting to not use dependencies

### PR DESCRIPTION
**Pull request by bergmania**
This PR refactor config/settings, to avoid using dependencies in Config/Settings.

For connectionstings:
- Changed to method injection of `IIOHelper` on the two methods to save and remvoe the connectionstring

For ModelsBuilder
- Handle the replace of `~\` with the virual path of the app outside of the settings

For GlobalSettings
- Moved the Path to IIOHelper

### CODE REVIEW

File: src/Umbraco.Configuration/ConfigsFactory.cs [hunk 1]
```diff
@@ -61,7 +58,6 @@ public Configs Create(IIOHelper ioHelper, ILogger logger)
             configs.Add<IMemberPasswordConfiguration>(() => MemberPasswordConfigurationSettings);
             configs.Add<IContentSettings>(() => ContentSettings);
 
-            configs.AddCoreConfigs(ioHelper);
```
Comment (elit0451) [inline src/Umbraco.Configuration/ConfigsFactory.cs]: Is it that for the moment we don't add the `GridConfig`?

File: src/Umbraco.Configuration/GlobalSettings.cs [hunk 2]
```diff
@@ -227,7 +214,7 @@ public string ConfigurationStatus
             }
             set
             {
-                SaveSetting(Constants.AppSettings.ConfigurationStatus, value, _ioHelper);
+                SaveSetting(Constants.AppSettings.ConfigurationStatus, value, Current.IOHelper); //TODO remove
```
Comment (elit0451) [inline src/Umbraco.Configuration/GlobalSettings.cs]: Is this TODO for later use?

File: src/Umbraco.Core/Configuration/GlobalSettingsExtensions.cs [hunk 3]
```diff
@@ -28,14 +28,15 @@ public static string GetUmbracoMvcArea(this IGlobalSettings globalSettings, IIOH
             return _mvcArea;
         }
 
+        //TODO Move to IOHelper
         internal static string GetUmbracoMvcAreaNoCache(this IGlobalSettings globalSettings, IIOHelper ioHelper)
```
Comment (elit0451) [inline src/Umbraco.Core/Configuration/GlobalSettingsExtensions.cs]: `globalSettings` is no longer needed and can be safely removed

File: src/Umbraco.Configuration/GlobalSettings.cs (hunk 2 above)
Comment (bergmania) [reply src/Umbraco.Configuration/GlobalSettings.cs]: This is a class that will be removed in the future.. I thing the TODO is fine for now

File: src/Umbraco.Configuration/ConfigsFactory.cs (hunk 1 above)
Comment (bergmania) [reply src/Umbraco.Configuration/ConfigsFactory.cs]: Yes.. `IGridConfig` is not a real config file as the rest.. It is just registered in a normal way.

**Pull request merged**